### ISSUE TITLE
Upcoming overhaul (now live in the master branch)

### ISSUE THREAD

**Issue by HikariKnight** [HikariKnight/material-awesome#13]
So as of the past half year i have been trying out PapyElGringo's material-shell and been running that along with system76's pop-shell, both of them i find lacking in one thing or the other.

So the other day i stumbled upon a fork of material-shell named [floppy-desktop](https://github.com/manilarome/the-glorious-dotfiles/tree/master/config/awesome/floppy) by manilarome, I liked a lot of the quality of life changes they had done to material-awesome while keeping things working and even updated for the upcoming awesome version.

They even added a notification center which i love, and fixed tons of bugs that were in material-awesome!

But i did not like the theme, the theme was replaced with something different.
So i am currently (with my limited to non existent lua knowledge) backporting the floppy-desktop to be more similar to the material-awesome it was forked from.

This upcoming overhaul will break support for awesome-stable (4.3 as of writing)
and will require users to compile awesome from source in order to get access to the new "ruled" library which manilarome has taken advantage of.

This means the current main branch will get archived as awesome4.3, while the new manilarome backported version will become the new main.

NOTE: all the things removed from floppy-desktop theme has been just commented out so whoever wants to use those features can re-enable them!

So far this is the progress visually
![image](https://user-images.githubusercontent.com/2557889/108844943-1118eb00-75dd-11eb-970b-71b0f1c53ce6.png)
![image](https://user-images.githubusercontent.com/2557889/108844997-29890580-75dd-11eb-8bd4-5cf8d15cd1d1.png)
![image](https://user-images.githubusercontent.com/2557889/108845040-373e8b00-75dd-11eb-8428-086e660ea652.png)
![image](https://user-images.githubusercontent.com/2557889/108845111-4f160f00-75dd-11eb-94b9-f653b8f14509.png)
![image](https://user-images.githubusercontent.com/2557889/108845146-589f7700-75dd-11eb-8aae-2add3afe180f.png)
![image](https://user-images.githubusercontent.com/2557889/108845195-6523cf80-75dd-11eb-8ed8-832c770dd22b.png)

For a comparison, this is what Floppy-Desktop looked like before i started "materializing" it
![image](https://user-images.githubusercontent.com/2557889/108845439-ba5fe100-75dd-11eb-968c-ee1094a24f17.png)

**Comment by eqan** [comment]
This looks fabulous hope we'll get more features in the form of scripts rather than bloat applications & some more tiling mods then what are present right now, I'm also trying my best to integerate features, but i think i need to go through lua documentation before i can get started, looking forward towards your progress @HikariKnight

**Comment by HikariKnight** [comment]
it is still using traditional applications, simply because it is what i use (so for me it is not bloat because it is functional and i use it ;) )
but from me playing around with it, it should be easier to integrate new modules.
However the window manager has a slower startup because of how heavily modified it is (12 seconds on my machine)

Like it will still be using rofi as launcher, it will still use ibus for keyboard (mainly because i use that along with uniemoji to write emojis when i need them), it will still use nm-applet because it does what i need, same with kdeconnect.

It is essentially my desktop setup people download so of course it will have applications i use in there.
however i am going to move more of the "dirty" startups into `~/.config/awesomestart´ so that those auto starts dont have to be included in the git repo, things like barrier/synergy, personal startup scripts like my pachannels to setup all my audio channels for things.

apps.lua will still include the startup things that work with awesome (and works after awesome is restarted with the hotkeys) and will also serve as an example on how to your own stuff.

picom will be used instead of compton, as compton does not fully support the new config and will crash while the screen is locked.

manilarome has done a lot of work on the backend for things like moving applications (that you have made rules for) to specific tags and such, however i copied out those rules and made a backup of them in a separate file as it did not play well with my triple monitor setup :smiley: so it will still be there in the source for anyone who wants to enable things like that.

In short this new config will require more manual labour in the form of having to compile 2 programs from source in order to use material-awesome2, and you will still have to configure it to fit what you personally use. it is not a simple git clone and done and probably never will be.

**Comment by eqan** [comment]
Currently i'm all in for better performance(mine currently suck around 300mbs of Ram) so i'm trying my best to replace these widgets with scripts from rofi or awesome window manager scripts wriiten in lua, It would be best if we could somehow integerate these scripts as applications replacmenets but to each their own, everybody has their own taste, still looking forward toward this

**Comment by HikariKnight** [comment]
Well this change is now live in the master branch and the old content is now archived to the awesome4.3-stable branch

current bugs (seems to be a bug with the new dbus notification api for awesome)
* Clicking notifications will just dismiss them, not raise the window/application
